### PR TITLE
0.2.3 -> 0.3.0 shims

### PR DESCRIPTION
**Pull request by bmcfee**
This PR adds deprecation warnings and some forward-looking compatibility for #149 .

Implements #152.

### CODE REVIEW

File: jams/core.py [hunk 1]
```diff
@@ -1168,6 +1195,25 @@ def slice(self, start_time, end_time, strict=False):
...
+            `intervals[i, :] = [time[i], time[i] + duration[i]]`
+
+        labels : list
+            List view of value field.
+        '''
+
+        times = timedelta_to_float(self.data.time.values)
+        duration = timedelta_to_float(self.data.duration.values)
+
+        return np.vstack([times, times + duration]).T, list(self.data.value)
```
Comment (ejhumphrey) [inline jams/core.py]: why `np.vstack` over `np.array`..?

File: jams/eval.py [hunk 2]
```diff
@@ -104,8 +104,8 @@ def beat(ref, est, **kwargs):
     namespace = 'beat'
     ref = coerce_annotation(ref, namespace)
     est = coerce_annotation(est, namespace)
-    ref_interval, _ = ref.data.to_interval_values()
-    est_interval, _ = est.data.to_interval_values()
+    ref_interval, _ = ref.to_interval_values()
```
Comment (ejhumphrey) [inline jams/eval.py]: ohgeez this is so much cleaner.

File: tests/jams_test.py [hunk 3]
```diff
@@ -211,10 +211,16 @@ def test_jamsframe_interval_values():
...
     jf = jams.JamsFrame.from_dataframe(df)
 
-    intervals, values = jf.to_interval_values()
+    warnings.resetwarnings()
+    warnings.simplefilter('always')
+    with warnings.catch_warnings(record=True) as out:
+        intervals, values = jf.to_interval_values()
+        assert len(out) > 0
+        assert out[0].category is DeprecationWarning
+        assert 'deprecated' in str(out[0].message).lower()
```
Comment (ejhumphrey) [inline tests/jams_test.py]: nice

File: jams/core.py (hunk 1 above)
Comment (bmcfee) [inline jams/core.py]: to concatenate them along the second axis; historical style holdover from the days before arbitary concat support.

but this is getting wiped in #149 also.

File: jams/core.py [hunk 4]
```diff
@@ -1168,6 +1202,34 @@ def slice(self, start_time, end_time, strict=False):
...
+
+    def __iter_obs__(self):
+        for _, (t, d, v, c) in self.data.iterrows():
```
Comment (ejhumphrey) [inline jams/core.py]: I find it odd that iterating `self` is (or will be) a shortcut to `self.data.obs` ... which feels like two unnecessary levels of abstraction? right?

Comment (bmcfee) [inline jams/core.py]: `self.data.obs` is now (will be) `self.data`

Comment (bmcfee) [inline jams/core.py]: ... but that's neither here nor there.  The point of this change is to add forward support for:

```python
for obs in annotation:
   # process jams.Observation object
```

the under-the-hood parts of how that iterator works won't matter.

Comment (ejhumphrey) [inline jams/core.py]: I guess I meant that `self -> Annotation`, `self.data -> AnnotationData`, `self.data.obs -> SortedListWithKey`, and so only the last contains the items yielded by the iterator.

Comment (ejhumphrey) [inline jams/core.py]: ah yes, i realize I forgot to contextualize my comment -- this change is fine; however, it occurs to me that the data structure is maybe deeper than necessary? not necessarily for iteration, but mutating it?

Comment (bmcfee) [inline jams/core.py]: Right -- check the changes I made to #149 last night / this morning.  That's all cleaned up now (and either way, irrelevant to this PR).

Comment (ejhumphrey) [inline jams/core.py]: ah yes, i see now that my understanding of things elsewhere is stale.

**Comment by bmcfee** [comment]
Aside from deprecation of `to_interval_values` and adding the `Observation` type and annotation iterator, is there anything else that we can do here to make the transition easier?

**Comment by bmcfee** [comment]
.. I'm inclined to think there isn't much else we can do here, since any other deprecation warnings would apply to methods called internally as well as the public.

Unless there are objections from @ejhumphrey , I think this is ready to merge and push as 0.2.3 release.

**Comment by ejhumphrey** [comment]
i have no other thoughts or opinions at the moment, tho I apologize in advance for when they occur to me in 4 days.

**Pull request merged**